interface/src/app/analyze/analyze.tpl.html

```diff
@@ -23,8 +23,8 @@ <h1 class="page-header">
             </div>
             <div class="media-body">
               <button type="button" class="btn"
-              ng-class="sb.has_item(search_item)?'disabled':''"
-              ng-click="sb.add_item(search_item)">
+              ng-class="sb.hasItem(search_item)?'disabled':''"
```

Comment:
In this case, I'd say all of the logic is in the `hasItem()` call and I'm just making use of it to specify the class. I like keeping the class name here in the HTML, if possible, rather than putting that into the controller logic. Does that separation make sense? Not sure I'm perfectly consistent on that, but this is what I eventually landed on as I worked through this code the first time.